gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ReplicationCopyMode.java

```diff
@@ -0,0 +1,44 @@
+package gobblin.data.management.copy.replication;
+
+import com.typesafe.config.Config;
+
+
+/**
+ * Specify the replication copy mode, either Pull or Push
+ * @author mitu
+ *
+ */
+public enum ReplicationCopyMode {
+  PULL("pull"),
+  PUSH("push");
```

Comment:
You may have to do some thinking about the push support with respect to the route selection. In push mode, it is a bit harder because different sources may try to make the decision at the same time. It is possible that some routing policies are not possible in push mode.